src/Arrays/MPIStateArrays.jl

```diff
@@ -417,24 +405,25 @@ quadrature weights from a grid, thus this becomes an integral approximation.
 function weightedsum(A::MPIStateArray, states=1:size(A, 2))
   isempty(A.weights) && error("`weightedsum` requires weights")
 
-  T = eltype(A)
+  FT = eltype(A)
   states = SVector{length(states)}(states)
 
   C = @view A.data[:, states, A.realelems]
   w = @view A.weights[:, :, A.realelems]
-  init = zero(DoubleFloat{T})
+  init = zero(DoubleFloat{FT})
```

Comment:
Why is `DoubleFloat` used here? To maintain accuracy? Also, what happens if `FT` is higher precision than `DoubleFloat`?